client/src/modules/journal/journal.js

```diff
@@ -77,9 +79,37 @@ function JournalController(Journal, Sorting, Grouping,
     flatEntityAccess           : true,
     enableGroupHeaderSelection : true,
     enableRowHeaderSelection   : true,
-    onRegisterApi : function (api) { vm.gridApi = api; }
+    onRegisterApi :  function onRegisterApi(api) {
+      vm.gridApi = api;
+      vm.gridApi.selection.on.rowSelectionChanged(null, rowSelectionChanged);
+      vm.gridApi.selection.on.rowSelectionChangedBatch(null, rowSelectionChanged);
+    },
+
   };
 
+  // row selection changed
+  function rowSelectionChanged() {
+    vm.selectedRows = vm.gridApi.selection.getSelectedGridRows();
+  }
+
+  // comment selected rows
+  vm.commentRows = function commentRows() {
+    Journal.openCommentModal({ rows : vm.selectedRows })
```

Comment:
Instead of using `vm.selectedRows`, try using the data in the `SelectionService`.  It should already be captured in the `JournalCtrl.selection.selected.groups` variable.  That way, you don't create expensive listeners.